test/minikube/minikube_long_restore_test.go

```diff
@@ -891,3 +891,220 @@ func TestKubernetesRestoreDatabaseSeparateJournal(t *testing.T) {
 		testlib.CheckRestoreRequests(t, namespaceName, admin0, opt.DbName, "", "")
 	})
 }
+
+// Extensive end to end tests for creating a database off of a VolumeSnapshot
+func TestCornerCaseKubernetesSnapshotRestore(t *testing.T) {
```

Comment:
How much time this test takes?

Reply:
Around 10 minutes on CircleCI. Is there some scenarios you would recommend cutting?

Reply:
Can you try using `t.Run("testSpecificScenario", ...)` so that test failures are easier to locate? The Golang testing framework doesn't have setup and teardown hooks, but you can define test cases in a hierarchical fashion in that way. Not sure if you would have to do something special in each test case to leave the environment in a state that the subsequent sub-tests can run despite a failure.